src/wp-includes/option.php

```diff
@@ -2037,6 +2037,25 @@ function get_network_option( $network_id, $option, $default_value = false ) {
 	 */
 	$pre = apply_filters( "pre_site_option_{$option}", false, $option, $network_id, $default_value );
 
+	/**
+	 * Filters the value of an existing network options before it is retrieved.
+	 *
+	 * Returning a truthy value from the filter will effectively short-circuit retrieval
+	 * and return the passed value instead.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param mixed  $pre         The value to return instead of the network option value. This differs
+	 *                            from `$default`, which is used as the fallback value in the event
+	 *                            the option doesn't exist elsewhere in get_network_option().
+	 *                            Default false (to skip past the short-circuit).
+	 * @param string $option      Name of the option.
```

Comment:
nit: In `pre_site_option_{$option}` this is documented as "Option name."  I think it would be for these two to be named the same.  I prefer this one, so maybe we change the other filter?